lib/ttn.js

```diff
@@ -0,0 +1,164 @@
+'use strict';
+
+// TODO: hook to register devices in TTN when called from osemApi or something
+// - return dev_uid, app_uid, app_key ?
+
+const { data, application, key } = require('ttn');
+const request = require('request-promise-native');
+
+const cfg = require('../config');
+const { boxFromDevId } = require('./database');
+const { TTNError } = require('./errors');
+const decoder = require('./decoding');
+const { createLogger } = require('./logging');
+
+const log = createLogger('ttn');
+
+const onUplinkMessage = function onUplinkMessage (dev_id, message) {
+  const { app_id, port } = message;
```

Comment:
Duplicate code to the `httpIntegrationHandler` ?

Reply:
This is because the response handling is different & I wanted to have different loggers for each protocol, and I didn't consider 10 duplicated lines to be worth an additional abstraction

Reply:
ok :)